modules/GEPanelApp.py

```diff
@@ -0,0 +1,248 @@
+
+* The OntoMa mapping process
+  ** https://github.com/opentargets/OnToma/blob/master/ontoma/interface.py#L365
+  ** Searches for a matching EFO code for a given phenotype/disease string
+     operations roughly ordered from least expensive to most expensive
+     and also from most authorative to least authorative
+        1. EFO OBO lookup
+        2. Zooma mappings lookup
+        3. Zooma API high confidence lookup
+        4. OLS API EFO lookup - exact match
+        --- below this line we might not have a term in the platform ---
+        5. HP OBO lookup
+        6. OLS API HP lookup - exact match
+        7. OLS API EFO lookup - not exact
+        (8. ?Zooma medium)
+
+'''
+
+#class GEPanelApp(GCSBucketManager):
+class GEPanelApp():
+    #ge_object = GE()
+    #ge_object.execute_ge_request()
+    #ge_object.use_zooma()
+    #ge_object.process_panel_app_file()
+    #ge_object.write_evidence_strings(Config.GE_EVIDENCE_FILENAME)
+
+    def __init__(self):
+        super(GEPanelApp, self).__init__()
+        self.panel_app_id_map = OrderedDict()
+        self.panel_app_info = list()
+        self.map_strings = OrderedDict()
+
+    def process_ge(self):
+
+        self.ontoma = OnToma()
+        gene_parser = GeneParser()
+        gene_parser._get_hgnc_data_from_json()
+
+        self.omim_to_efo_map = self.ontoma._omim_to_efo
+        self.symbol_to_ensembl_map = gene_parser.genes
+
+        print("Parsed %s OMIM to EFO mapping " % len(self.omim_to_efo_map))
+        print("Parsed %s ZOOMA to EFO mapping " % len(self.ontoma._zooma_to_efo_map))
+        print("Parsed %s Name to EFO mapping " % len(self.ontoma.name_to_efo))
+        print("Parsed %s Name to HP mapping " % len(self.ontoma.name_to_hp))
+        self.get_panel_code_mapping()
+        self.process_ge_panel()
+
+    def get_panel_code_mapping(self, filename=Config.GE_PANEL_MAPPING_FILENAME):
+
+        print('Mapping old GE panel id to new panel id, using : ' + filename)
+
+        #with open(self.get_gcs_filename(filename), mode='rt') as fh:
+        with open(filename, mode='rt') as fh:
+            reader = csv.reader(fh, delimiter=',', quotechar='"')
+            next(reader, None) # skip header
+
+            for row in reader:
+                (old_panel_id, new_panel_id) = row
+                print ("panel id :%s => is now %s"%(old_panel_id, new_panel_id))
+                self.panel_app_id_map[old_panel_id] = new_panel_id
+
+    def process_ge_panel(self):
+        '''
+         Create panel app info list and phenotype set
+         :return: Unique phenotype list
+        '''
+        print("Retrieving GE PanelApp data...")
+        url = 'https://panelapp.genomicsengland.co.uk/WebServices/search_genes/all/'
+        phenotype_list = []
+        c = 0
+        d = 0
+        e = 0
+        f = 0
+
+        for panel_name, panel_id, panel_version, panel_diseasegroup, panel_diseasesubgroup in self.get_panel_list():
```

Comment:
why don't you declare a `Panel` class? it might allow you to iterate through `Panel` objects more succinctly and parse each object with code kept outside of the request handling logic.